app/Http/Controllers/Api/UserInvitation/Services/UserInvitationService.php

```diff
@@ -0,0 +1,63 @@
+<?php
+namespace App\Http\Controllers\Api\UserInvitation\Services;
+
+// Models
+
+use App\User;
+use App\Models\UserInvitation;
+use App\DomainValueObjects\UUIDGenerator\UUID;
+// Contracts
+use App\Http\Controllers\Api\UserInvitation\Contracts\UserInvitationContract;
+// Exceptions
+
+
+use App\Exceptions\UserInvitationExceptions\UserInviteCreationFailed;
+use App\Exceptions\UserInvitationExceptions\UserAlreadyRegistered;
+use Dirape\Token\Token as DirapeToken;
+
+
+class UserInvitationService implements UserInvitationContract
+{
+    public function inviteNewUser($organizationId, $roleId, $name, $email): array
+    {
+         $this->deletePreviouslyCreatedUserInvitation($email);
+        if(!$this->verifyUserIsNotAlreadyRegistered($email)) {
```

Comment:
I think this if check should come before deleting the previouslycreatedUserInvitation.

Reply:
Done, thank you.